src/features/administrative/scripts/import-statistics.ts

```diff
@@ -10,83 +10,80 @@
-    `/Location/?identifier=${identifier}`
-  )
-  const location: fhir.Location =
-    locationSearchResult &&
-    locationSearchResult.entry &&
-    locationSearchResult.entry[0] &&
-    locationSearchResult.entry[0].resource
-  return location
-}
-
-function generateStatisticalExtensions(sourceStatistic: LocationStatistic) {
-  const malePopulations = []
-  const femalePopulations = []
-  const totalPopulations = []
-  const birthRates = []
-
-  for (const year of sourceStatistic.years) {
-    femalePopulations.push({
-      [year.year]: year.female_population
-    })
-    malePopulations.push({
-      [year.year]: year.male_population
-    })
-    totalPopulations.push({
-      [year.year]: year.population
-    })
-    birthRates.push({
-      [year.year]: year.crude_birth_rate / 2
-    })
-  }
-
-  const extensions: fhir.Extension[] = [
-    {
-      url: 'http://opencrvs.org/specs/id/statistics-male-populations',
-      valueString: JSON.stringify(malePopulations)
-    },
-    {
-      url: 'http://opencrvs.org/specs/id/statistics-female-populations',
-      valueString: JSON.stringify(femalePopulations)
-    },
-    {
-      url: 'http://opencrvs.org/specs/id/statistics-total-populations',
-      valueString: JSON.stringify(totalPopulations)
-    },
-    {
-      url: 'http://opencrvs.org/specs/id/statistics-crude-birth-rates',
-      valueString: JSON.stringify(birthRates)
-    }
-  ]
-  return extensions
-}
+import { readCSVToJSON, sendToFhir } from '@countryconfig/features/utils'
+import { Location } from '@countryconfig/scripts/validate-source-files'
+import { z } from 'zod'
+import {
+  extractLocationTree,
+  extractMaxAdminLevel,
+  extractStatisticsMap,
+  generateStatisticalExtensions,
+  getChildLocations,
+  getLocationByIdentifier,
+  getStatistics,
+  mergeLocationStatistics
+} from './utils'
 
 async function addStatisticalData() {
-  const statistics = await getStatistics(process.argv[2])
-  // tslint:disable-next-line:no-console
+  const stats = await getStatistics(process.argv[2])
+  const rawLocations = (await readCSVToJSON(process.argv[3])) as Array<
+    z.infer<typeof Location>
+  >
+  const maxAdminLevel = extractMaxAdminLevel(rawLocations as any)
+  const locationMap = extractLocationTree(rawLocations as any, maxAdminLevel)
+  const statisticMap = extractStatisticsMap(stats)
+
+  // eslint-disable-next-line no-console
   console.log(
     `${chalk.blueBright(
       '/////////////////////////// UPDATING LOCATIONS WITH STATISTICAL DATA IN FHIR ///////////////////////////'
     )}`
   )
 
-  for (const statistic of statistics) {
+  const alreadyUpdatedLocations: string[] = []
+
+  // adminLevel 0 doesn't need to be added as it's the country level
+  // If it needs to be added in the future, amend the condition to be `adminLevel >= 0`
+  for (let adminLevel = maxAdminLevel; adminLevel > 0; adminLevel--) {
+    for (let i = 0; i < rawLocations.length; i++) {
+      const column = `admin${adminLevel}Pcode`
```

Comment:
So column is equal to the lowest child level right?

Reply:
at first yeah,  it's `admin3Pcode` for example if 3 is the smallest level of administration. Then it tries to find the statistics for it with `let statisticsForLocation = statisticMap.get(location)`. If none are found, it will find the child locations. For smallest level of admin, we always have stats (due to validation ensuring that), so essentially `statisticsForLocation` would never be falsy for `admin3Pcode`.